vm/parser.ts

```diff
@@ -711,6 +861,17 @@ module J2ME {
       }
     }
 
+    getClassNameSlow(): string {
+      return this._name || (this._name = ByteStream.readString(this.utf8Name));
+    }
```

Comment:
Can you put a getter on the proto and a direct property on the object? After the first access it should be super fast.